### PR TITLE
Raise error when default configs have too wide permissions

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Raising error when default configs (config.toml or connections.toml) have too wide permissions. Error will not be raised if some other config is passed even if defaults have too wide permissions.

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -187,3 +188,38 @@ def test_connections_toml_override_config_toml(test_snowcli_config, snowflake_ho
     assert CONFIG_MANAGER["connections"] == {
         "default": {"database": "overridden_database"}
     }
+
+
+def test_too_wide_permissions_on_config_file(snowflake_home: Path):
```
Comment (sfc-gh-pjob) [inline tests/test_config.py:193]: Can we add a test parameter representing chmod to test various chmods?

File: tests/test_config.py [hunk 2]
```diff
@@ -187,3 +188,38 @@ def test_connections_toml_override_config_toml(test_snowcli_config, snowflake_ho
     assert CONFIG_MANAGER["connections"] == {
         "default": {"database": "overridden_database"}
     }
+
```
Comment (sfc-gh-pjob) [inline tests/test_config.py:191]: Can we add one simple test case (probably in some other test suite) confirming that running an actual command will fail (not only config initialization)?

File: src/snowflake/cli/api/secure_utils.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+def file_permissions_are_strict(file_path: Path) -> bool:
+    accessible_by_others = (
+        # https://docs.python.org/3/library/stat.html
+        stat.S_IRGRP  # readable by group
+        | stat.S_IROTH  # readable by others
+        | stat.S_IWGRP  # writeable by group
+        | stat.S_IWOTH  # writeable by others
+        | stat.S_IXGRP  # executable by group
+        | stat.S_IXOTH  # executable by group
```
Comment (sfc-gh-mraba) [inline src/snowflake/cli/api/secure_utils.py:13]:
```suggestion
        | stat.S_IXOTH  # executable by others
```

**Review by sfc-gh-mraba — APPROVED** [review]

**Review by sfc-gh-pczajka — APPROVED** [review]

File: src/snowflake/cli/api/config.py [hunk 4]
```diff
@@ -188,3 +192,10 @@ def _get_envs_for_path(*path) -> dict:
 def _dump_config(conf_file_cache: Dict):
     with open(CONFIG_MANAGER.file_path, "w+") as fh:
         dump(conf_file_cache, fh)
+
+
+def _check_config_files_permissions() -> None:
```
[the remark this answers is not in the document]
Comment (sfc-gh-astus) [reply src/snowflake/cli/api/config.py:197]: Changed

File: tests/test_sql.py [hunk 5]
```diff
@@ -107,3 +108,22 @@ def test_sql_overrides_connection_configuration(mock_conn, runner, mock_cursor):
         password="passFromTest",
         mfa_passcode=None,
     )
+
+
+def test_fail_command_when_default_config_has_too_wide_permissions(
```
Comment (sfc-gh-pjob) [inline tests/test_sql.py:113]: Maybe `test_main` or `test_cli` would be a better location for this test case. WDYT?

Comment (sfc-gh-astus) [reply tests/test_sql.py]: Moved to test_main

**Pull request merged**